### PR TITLE
Get rid of Neutron PHP Standard

### LINKED ISSUE #65: [Feature Request]: Get rid of Neutron PHP Standard
### Is your feature request related to a problem?

[Neutron PHP Standard](https://github.com/Automattic/phpcs-neutron-standard) is a coding standard for PHP Code Sniffer. It was developed by Automattic but it is now abandoned by ~2 years.

As they state in their readme, they abandoned it because wanted to fully switch to [Vip Coding Standards](https://github.com/Automattic/VIP-Coding-Standards/), however it is not clear of all the roles from the old package have been moved.

### Describe the desired solution

These are the rules coming from Neutron that are used in this package:
- NeutronStandard.AssignAlign.DisallowAssignAlign
- NeutronStandard.Functions.DisallowCallUserFunc
- NeutronStandard.Globals.DisallowGlobalFunctions
- NeutronStandard.MagicMethods.DisallowMagicSerialize
- NeutronStandard.StrictTypes.RequireStrictTypes
- NeutronStandard.Whitespace.DisallowMultipleNewlines
- NeutronStandard.Whitespace.RequireNewlineBetweenFunctions

We need to check one by one if they have already a replacement in VIP Coding Standards, if not, we should probably write a custom one for them.

### Describe the alternatives that you have considered

I cannot see any alternative solution.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gmazzap**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

**What is the current behavior?** (You can also link to an open issue here)

See #65 

**What is the new behavior (if this is a feature change)?**

- Added custom `DisableCallUserFuncSniff` (with tests)
- Added custom `DisableMagicSerializeSniff` (with tests)
- Added custom `NoRootNamespaceFunctionsSniff` (with tests)
- Added Slevomat standard as dependency
- Explicilty required PHPCSExtra and PHPCS Utils
- Replaced some Neutron sniffs with Generic/Squid/PHPCSExtra/Slevomat
- Renamed updated WPCS standards
- Improved QA

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

Yes. Any configuration or ignore/disable comments referring Neutron standard will break

**Other information**:

### CODE REVIEW

File: tests/fixtures/disable-call-user-func.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+echo 'call_user_func_array';
+
+$foo = [
+    'call_user_func',
+    'call_user_func_array',
+];
+
+class Foo {
+
+    pribate function test() {
```
Comment (tfrommen) [inline tests/fixtures/disable-call-user-func.php]:
```suggestion
    private function test() {
```

File: Inpsyde/Sniffs/CodeQuality/DisableCallUserFuncSniff.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        // phpcs:enable Inpsyde.CodeQuality.NoAccessors
+        return [
+            'call_user_func' => [
+                'type' => 'error',
+                'message' => 'Usage of %s() is forbidden.',
+                'functions' => [
+                    'call_user_func',
+                    'call_user_func_array',
+                    'suca',
```
Comment (tfrommen) [inline Inpsyde/Sniffs/CodeQuality/DisableCallUserFuncSniff.php]: What is this? 🤔

Comment (gmazzap) [reply Inpsyde/Sniffs/CodeQuality/DisableCallUserFuncSniff.php]: Sorry, debug leftover

Comment (antonioeatgoat) [reply Inpsyde/Sniffs/CodeQuality/DisableCallUserFuncSniff.php]: 😂 😂 😂 😂

File: tests/fixtures/disallow-magic-serialize.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+// @phpcsSniff CodeQuality.DisableMagicSerialize
+
+class Foo {
```
Comment (antonioeatgoat) [inline tests/fixtures/disallow-magic-serialize.php:5]: What about `__unserialize` and `__wakeup`?

**Review by antonioeatgoat — COMMENTED** [review]
All fine for me, just noticed a very little thing that could be added.

File: tests/fixtures/disallow-magic-serialize.php (hunk 3 above)
Comment (gmazzap) [reply tests/fixtures/disallow-magic-serialize.php:5]: @antonioeatgoat those are not in the fixture, but are in the sniff: https://github.com/inpsyde/php-coding-standards/blob/a385aee01e8a6d8870fcb625627db70fbef034c0/Inpsyde/Sniffs/CodeQuality/DisableMagicSerializeSniff.php#L34-L37 

Considering how the sniff works, with the disabled funcions in a single array, if a couple works, the other two should work as well. 

Anyway, in [#eb1a128](https://github.com/inpsyde/php-coding-standards/commit/eb1a128afb68e161929b701ec1da38c33c2f8f9c) I added more tests :)

**Pull request merged**